l1-contracts/src/core/Rollup.sol

```diff
@@ -108,5 +115,19 @@ contract Rollup is IRollup {
     if (ts < lastBlockTs) {
       revert Errors.Rollup__TimestampTooOld();
     }
+
+    // Assert that the last blocks eth hash is included in the block
+    // For the first block we use 0.
+    uint256 lastEthBlockHash;
+    uint256 ethBlockHash;
+    assembly {
+      ethBlockHash := or(shl(128, ethBlockHashHi), ethBlockHashLo)
+    }
+    if (lastBlockNumber != 0) {
+      lastEthBlockHash = uint256(blockhash(lastBlockNumber));
```

Comment:
This is very error prone. Cannot access all historic states only the last 256. Why not store the blockhash instead of the last block number? 

You have the lastblockTs which you can use for the `!= 0 ` so not sure why you would not just store the `lastEthBlockHash` directly?

Reply:
As mentioned in our offline discussion, ill update this such that we end up storing this block -1 such that a block can always be provided